### PR TITLE
Extend RelatedEntitiesCard usages to allow custom columns

### PR DESCRIPTION
**Pull request by blakestoddard**
The `Depends on [components/resources]` tables on the entity detail page have been a source of questions from users for us because a) the naming (component/resource) doesn't match how we talk internally, and b) they showed columns that weren't entirely relevant to how we wanted to use them (in our case for resources as an example, we don't care about owner or lifecycle, but we do care about a custom compliance attribute). This allows you to specify a set of `TableColumns` via component props rather than needing to rebuild the component to support that.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog/src/components/DependsOnResourcesCard/DependsOnResourcesCard.tsx [hunk 1]
```diff
@@ -27,17 +27,23 @@ import {
 /** @public */
 export interface DependsOnResourcesCardProps {
   variant?: InfoCardVariants;
+  title?: string;
```
Comment (blakestoddard) [inline plugins/catalog/src/components/DependsOnResourcesCard/DependsOnResourcesCard.tsx:30]: Matches the `DependsOnResourcesCard` prop list to `DependsOnComponentsCard`, which already supported an optional title prop.

**Comment by freben** [comment]
Note the build output saying:

```
*************************************************************************************
* You have uncommitted changes to the public API of a package.                      *
* To solve this, run `yarn build:api-reports` and commit all api-report.md changes. *
*************************************************************************************
```

**Comment by blakestoddard** [comment]
:wave: Hey friends! Can you reopen this? I rebased and pushed the api-report changes.

**Comment by blakestoddard** [comment]
Bumpin this @freben

**Comment by freben** [comment]
Oh i see now - reopening actually doesn't work after a force push.

**Comment by blakestoddard** [comment]
Ah, neat TIL -- I'll get a new one opened up. Thanks!

**Pull request closed without merging**